build_tools/bump_submodules.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python
+
+import argparse
+from pathlib import Path
+from datetime import datetime
+import shlex
+import subprocess
+import sys
+
+THIS_SCRIPT_DIR = Path(__file__).resolve().parent
+THEROCK_DIR = THIS_SCRIPT_DIR.parent
+
+
+def log(*args, **kwargs):
+    print(*args, **kwargs)
+    sys.stdout.flush()
+
+
+def exec(args: list[str | Path], cwd: Path):
+    args = [str(arg) for arg in args]
+    log(f"++ Exec [{cwd}]$ {shlex.join(args)}")
+    sys.stdout.flush()
+    subprocess.check_call(args, cwd=str(cwd), stdin=subprocess.DEVNULL)
+
+
+def pin_tensile():
+    tag_file_path = THEROCK_DIR / "math-libs" / "BLAS" / "rocBLAS" / "tensile_tag.txt"
+    with open(tag_file_path) as tag_file:
+        tensile_tag = tag_file.read().splitlines()
+
+    exec(
+        ["git", "checkout", tensile_tag[0]],
+        cwd=THEROCK_DIR / "math-libs" / "BLAS" / "Tensile",
+    )
+
+
+def run(args):
+    date = datetime.today().strftime("%Y%m%d")
+
+    if args.create_branch or args.push_branch:
+        exec(
+            ["git", "checkout", "-b", args.branch_name],
+            cwd=THEROCK_DIR,
+        )
+
+    exec(
+        ["./build_tools/fetch_sources.py", "--remote", "--no-apply-patches"],
+        cwd=THEROCK_DIR,
+    )
+
+    if args.pin_tensile:
+        pin_tensile()
+
+    exec(
+        ["git", "commit", "-a", "-m", "Bump submodules " + date],
+        cwd=THEROCK_DIR,
+    )
+
+    try:
+        subprocess.check_call(
+            "./build_tools/fetch_sources.py", cwd=THEROCK_DIR, stdin=subprocess.DEVNULL
+        )
+    except subprocess.CalledProcessError as patching_error:
+        log(f"Failed to apply patches")
+        sys.exit(1)
+
+    if args.push_branch:
+        exec(
+            ["git", "push", "-u", "origin", +args.branch_name],
```

Comment:
what's the `+` doing here?
```suggestion
            ["git", "push", "-u", "origin", args.branch_name],
```

Reply:
I think that was just a typo.